src/registryctl/api/registry/manifest/manifest.go

```diff
@@ -34,25 +41,37 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 // delete deletes manifest ...
 func (h *handler) delete(w http.ResponseWriter, r *http.Request) {
+	var span trace.Span
+	ctx := r.Context()
 	ref := mux.Vars(r)["reference"]
+	if tracelib.Enabled() {
+		ctx, span = tracelib.StartTrace(ctx, tracerName, "delete-manifest", trace.WithAttributes(attribute.Key("method").String(r.Method)))
```

Comment:
ctx is overwrited here, is it expected?

Reply:
yes, it added trace info in ctx,  if related code in docker distribution has to trace logic, it will link its span to harbor's span inside one trace. And it's not harmful if docker distribution don't need such trace info.